### ISSUE TITLE
Inheritance Effector adding a curve option

### ISSUE THREAD

**Issue by kdsin23** [harisreedhar/AN_BLUEFOX#8]
![shot3](https://user-images.githubusercontent.com/64368228/80444654-c7239200-8944-11ea-8b57-46ad708b4c48.png)

my question is  how to make the points move along the path from sphere to plane? as far as i know ,in C4D we can change the path of the point use Inheritance Effector.
is it possible to add a curve option to your node,if sucess,that would be great! 
Thank you in advanced!

**Comment by harisreedhar** [comment]
That would be great. But i don't know how would they work, do you have any idea to implement that using built in nodes?

**Comment by kdsin23** [comment]
> That would be great. But i don't know how would they work, do you have any idea to implement that using built in nodes?

my idea is that use "evaluate Spline Node"  to cacluate locations of points,

![inheriance effector test](https://user-images.githubusercontent.com/64368228/80466358-07980580-896f-11ea-94ec-327cac0c2106.gif)

so far,it can be added a single empty object to control the path, but i don't know how to add more empty object to control path dynamically.

**Comment by harisreedhar** [comment]
> > That would be great. But i don't know how would they work, do you have any idea to implement that using built in nodes?
> 
> my idea is that use "evaluate Spline Node" to cacluate locations of points,
> 
> ![inheriance effector test](https://user-images.githubusercontent.com/64368228/80466358-07980580-896f-11ea-94ec-327cac0c2106.gif)
> 
> so far,it can be added a single empty object to control the path, but i don't know how to add more empty object to control path dynamically.

Excellent idea bro. Only small changes are needed to implement curve:
![curveinheritance](https://user-images.githubusercontent.com/46858047/80508047-2c9f6f00-8995-11ea-9f75-2fc364430dfa.gif)

**Comment by harisreedhar** [comment]
Prototype ready 👍 . Now i have to find ways to improve and clean the code. Thank you so much for the idea.
![prototypecurve](https://user-images.githubusercontent.com/46858047/80524038-9e36e780-89ac-11ea-9c0f-1698f9f3822c.gif)

**Comment by harisreedhar** [comment]
Added Falloff 

![prototypeFalloff](https://user-images.githubusercontent.com/46858047/80528412-857e0000-89b3-11ea-881e-0d654abb46e4.gif)

**Comment by kdsin23** [comment]
> Excellent idea bro. Only small changes are needed to implement curve:
> ![curveinheritance](https://user-images.githubusercontent.com/46858047/80508047-2c9f6f00-8995-11ea-9f75-2fc364430dfa.gif)

use a "evaluate Spline Node" to cacluate middle points instead of "Spline info Node", using "Spline info Node", we must subdivide the spline to add points, but "evaluate Spline Node" we can control the count to add points counts, may be i am wrong.

![shot1](https://user-images.githubusercontent.com/64368228/80554777-c8b29000-8a00-11ea-8dc0-5e304f1a781a.png)

**Comment by kdsin23** [comment]
> Added Falloff
> 
> ![prototypeFalloff](https://user-images.githubusercontent.com/46858047/80528412-857e0000-89b3-11ea-881e-0d654abb46e4.gif)

oh,it is nice! I was excited about this ,expecting:)

**Comment by harisreedhar** [comment]
> > Excellent idea bro. Only small changes are needed to implement curve:
> > ![curveinheritance](https://user-images.githubusercontent.com/46858047/80508047-2c9f6f00-8995-11ea-9f75-2fc364430dfa.gif)
> 
> use a "evaluate Spline Node" to cacluate middle points instead of "Spline info Node", using "Spline info Node", we must subdivide the spline to add points, but "evaluate Spline Node" we can control the count to add points counts, may be i am wrong.
> 
> ![shot1](https://user-images.githubusercontent.com/64368228/80554777-c8b29000-8a00-11ea-8dc0-5e304f1a781a.png)
> 
> 
> 
> 
> 

Do we really need that? I think it's users responsibility to provide a spline with proper subdivision, also evaluate range adds more computations. what do you think?

**Comment by kdsin23** [comment]
yes,I agree with you! later on, i think using "spline info node" is better.
Prototype version of curve inheritance effector is coming.
thank you!

**Comment by harisreedhar** [comment]
> yes,I agree with you! later on, i think using "spline info node" is better.
> Prototype version of curve inheritance effector is coming.
> thank you!

I tried to cythonize the code, but not much speed improvement, for 50000 points it took 1sec to calculate. Faster than inbuilt nodes but still i'm not happy 😌. Did you tried yourself ?

![cythoninheritance](https://user-images.githubusercontent.com/46858047/80582744-dda51880-8a2c-11ea-9042-1837976295e3.png)

**Comment by kdsin23** [comment]
> > yes,I agree with you! later on, i think using "spline info node" is better.
> > Prototype version of curve inheritance effector is coming.
> > thank you!
> 
> I tried to cythonize the code, but not much speed improvement, for 50000 points it took 1sec to calculate. Faster than inbuilt nodes but still i'm not happy relieved. Did you tried yourself ?


sorry,I am not familiar with cythonize,I think you may ask this issue on official branch,maybe someone can fix this problem!

**Comment by harisreedhar** [comment]
> > > yes,I agree with you! later on, i think using "spline info node" is better.
> > > Prototype version of curve inheritance effector is coming.
> > > thank you!
> > 
> > I tried to cythonize the code, but not much speed improvement, for 50000 points it took 1sec to calculate. Faster than inbuilt nodes but still i'm not happy relieved. Did you tried yourself ?
> 
> 
> sorry,I am not familiar with cythonize,I think you may ask this issue on official branch,maybe someone can fix this problem!
> 
> 
I think evaluate spline causes the slow down

**Comment by harisreedhar** [comment]
Added this feature in 3e9c5aaa788583d3a7aaa840eda55d6a6c4e766e . Lot of optimization is still pending, feel free to reopen this issue for further improvements.
